### PR TITLE
Fix asset resource request not properly cleaning up

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

File: libraries/networking/src/AssetClient.cpp [hunk 1]
```diff
@@ -474,15 +473,53 @@ bool AssetClient::renameAssetMapping(const AssetPath& oldPath, const AssetPath&
...
+
+bool AssetClient::cancelUploadAssetRequest(MessageID id) {
+    // Search through each pending mapping request for id `id`
+    for (auto& kv : _pendingUploads) {
+        if (kv.second.erase(id)) {
+            return true;
+        }
+    }
+    return false;
+}
```
Comment (huffman) [inline libraries/networking/src/AssetClient.cpp]: I can DRY these cancel requests up in a future PR. It requires some other unrelated refactoring work and I'd prefer that this fix gets in first.

**Comment by huffman** [comment]
This is for https://app.asana.com/0/26224634468389/107178512251708

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e375fd076b57350a6e535d6896860df68821c3e available at: https://gist.github.com/201ae31dbf2815a24411a0751cee88ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7539.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7539.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0948c8cefb2eeaa4a5395f24d9aba2541708b95c available at: https://gist.github.com/201ae31dbf2815a24411a0751cee88ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7539.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7539.dmg

**Comment by ericrius1** [comment]
I am no longer crashing in forum with this build

**Pull request merged**